packages/world/src/systems/SpawnSystem.sol

```diff
@@ -93,6 +101,7 @@ contract SpawnSystem is System {
   }
 
   function _spawnPlayer(uint32 playerMass, VoxelCoord memory spawnCoord) internal returns (EntityId) {
+    // TODO: check that it is revealed
```

Comment:
I think this already happens since if the entity exists, its already revealed, otherwise it calls `TerrainLib._getBlockType` which reverts if not revealed